// @jshookmcp/extension-sdk/dist/plugin.js
import { existsSync } from 'node:fs';
import { dirname, join } from 'node:path';
import { fileURLToPath } from 'node:url';
import dotenv from 'dotenv';
const loadedEnvPaths = new Set();
export function loadPluginEnv(manifestUrl) {
    const pluginDir = dirname(fileURLToPath(manifestUrl));
    const envPath = join(pluginDir, '.env');
    if (loadedEnvPaths.has(envPath))
        return;
    if (existsSync(envPath)) {
        dotenv.config({ path: envPath, override: false });
    }
    loadedEnvPaths.add(envPath);
}
function normalizeSegment(value) {
    return value.trim().replace(/[^a-zA-Z0-9]+/g, '_').replace(/^_+|_+$/g, '').toUpperCase();
}
function envCandidates(pluginId, key) {
    const pluginSegment = normalizeSegment(pluginId);
    const keySegment = normalizeSegment(key);
    return [
        `PLUGIN_${pluginSegment}_${keySegment}`,
        `PLUGINS_${pluginSegment}_${keySegment}`,
    ];
}
function parseBoolean(raw) {
    if (raw == null)
        return undefined;
    const value = raw.trim().toLowerCase();
    if (['1', 'true', 'yes', 'on'].includes(value))
        return true;
    if (['0', 'false', 'no', 'off'].includes(value))
        return false;
    return undefined;
}
export function getPluginBooleanConfig(ctx, pluginId, key, fallback) {
    for (const candidate of envCandidates(pluginId, key)) {
        const parsed = parseBoolean(process.env[candidate]);
        if (parsed !== undefined)
            return parsed;
    }
    return ctx.getConfig(`plugins.${pluginId}.${key}`, fallback);
}
const VALID_BOOST_TIERS = new Set([
    'search',
    'minimal',
    'workflow',
    'full',
    'reverse',
]);
export function getPluginBoostTier(pluginId) {
    for (const candidate of envCandidates(pluginId, 'BOOST_DOMAIN')) {
        const raw = process.env[candidate]?.trim().toLowerCase();
        if (raw && VALID_BOOST_TIERS.has(raw))
            return raw;
    }
    const globalDefault = process.env.MCP_DEFAULT_PLUGIN_BOOST_TIER?.trim().toLowerCase();
    if (globalDefault && VALID_BOOST_TIERS.has(globalDefault))
        return globalDefault;
    return 'full';
}
